packages/dids/src/methods/did-dht.ts

```diff
@@ -1025,6 +1035,16 @@ export class DidDhtDocument {
           // Convert the public key from a byte array to JWK format.
           let publicKey = await DidDhtUtils.keyConverter(namedCurve).bytesToPublicKey({ publicKeyBytes });
 
+          // DID DHT spec requires `alg` in keys in the DID document
+          publicKey.alg = parsedAlg || getJoseSignatureAlgorithmFromPublicKey(publicKey);
+
+          // Determine the Key ID (kid): '0' for the identity key or JWK thumbprint for others.
+          if (id === '0') {
+            publicKey.kid = '0';
+          } else if (publicKey.kid === undefined) {
+            publicKey.kid = await computeJwkThumbprint({ jwk: publicKey });
+          }
```

Comment:
@decentralgabe just confirmed that `id` MUST be `0` for identity key, and `undefined` otherwise, (we should have tests if they don't exist to make sure these are true in implementation), so it seems to me the condition can simply be:

```suggestion
          if (id === undefined) {
            publicKey.kid = await computeJwkThumbprint({ jwk: publicKey });
          }
```

Happy to discuss if I am not making sense!

Reply:
i'm going to update the spec language to always remove the id property
@thehenrytsai correctly pointed out we can discern the 0 key based on the did identifier
all other keys will use the thumbprint